### PR TITLE
feat: [CDS-44750]: env infra as expression

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
### Summary

Add support for expression for single env and single infrastructure under environment tab
Support for expression in run pipeline form will be added as part of separate ticket

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx [hunk 1]
```diff
@@ -377,7 +384,7 @@ export default function DeployEnvironment({
             selectItems={selectOptions}
           />
         )}
-        {isFixed && !isUnderEnvGroup && (
+        {isFixed && !isUnderEnvGroup && !isExpression && (
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx:387]:
```suggestion
        {isFixed && !isUnderEnvGroup && (
```

File: src/modules/70-pipeline/utils/runPipelineUtils.ts [hunk 2]
```diff
@@ -420,3 +420,11 @@ export function getFilteredAllowableTypes(allowableTypes: AllowedTypes, viewType
     ? allowableTypes
     : ((allowableTypes as MultiTypeInputType[]).filter(allowedType => !isMultiTypeRuntime(allowedType)) as AllowedTypes)
 }
+
+export function getAllowableTypesWithoutFixedValue(allowableTypes: MultiTypeInputType[]): AllowedTypes {
+  return allowableTypes.filter(type => isMultiTypeFixed(type)) as AllowedTypes
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/utils/runPipelineUtils.ts:425]:
```suggestion
  return allowableTypes.filter(type => !isMultiTypeFixed(type)) as AllowedTypes
```

File: src/modules/70-pipeline/utils/runPipelineUtils.ts [hunk 3]
```diff
@@ -420,3 +420,11 @@ export function getFilteredAllowableTypes(allowableTypes: AllowedTypes, viewType
...
+}
+
+export function getAllowableTypesWithoutExpression(allowableTypes: MultiTypeInputType[]): AllowedTypes {
+  return allowableTypes.filter(type => isMultiTypeExpression(type)) as AllowedTypes
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/utils/runPipelineUtils.ts:429]:
```suggestion
  return allowableTypes.filter(type => !isMultiTypeExpression(type)) as AllowedTypes
```

**Review by rameshx — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good @AshwinBhatkal-Harness

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**